gradle/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -0,0 +1,200 @@
+        if ((new File(exePath)).exists()) {
+            return exePath;
+        }
+        if ((new File(exePath + ".exe")).exists()) {
+            return exePath + ".exe";
+        }
+
+        // next up try a packaged version of sentry-cli
+        def cliSuffix
+        def osName = System.getProperty("os.name").toLowerCase()
+        if (osName.indexOf("mac") >= 0) {
+            cliSuffix = "Darwin-x86_64"
+        } else if (osName.indexOf("linux") >= 0) {
+            cliSuffix = "Linux-" + System.getProperty("os.arch")
+        } else if (osName.indexOf("win") >= 0) {
+            cliSuffix = "Windows-i686.exe"
+        }
+
+        if (cliSuffix != null) {
+            def resPath = "/bin/sentry-cli-${cliSuffix}";
+            def fsPath = SentryPlugin.class.getResource(resPath).getFile()
+
+            // if we are not in a jar, we can use the file directly
+            if ((new File(fsPath)).exists()) {
+                return fsPath;
+            }
+
+            // otherwise we need to unpack into a file
+            def resStream = SentryPlugin.class.getResourceAsStream(resPath)
+            File tempFile = File.createTempFile(".sentry-cli", ".exe")
+            tempFile.deleteOnExit()
+            def out = new FileOutputStream(tempFile);
+            try {
+                IOUtils.copy(resStream, out)
+            } finally {
+                out.close()
+            }
+            tempFile.setExecutable(true)
+            return tempFile.getAbsolutePath()
+        }
+
+        return "sentry-cli";
+    }
+
+    /**
+     * Returns the proguard task for the given project and variant.
+     *
+     * @param project
+     * @param variant
+     * @return
+     */
+    static Task getProguardTask(Project project, ApplicationVariant variant) {
+        def name = variant.name.capitalize()
+        def rv = project.tasks.findByName("transformClassesAndResourcesWithProguardFor${name}")
+        if (rv != null) {
+            return rv
+        }
+        return project.tasks.findByName("proguard${name}")
+    }
+
+    /**
+     * Returns the dex task for the given project and variant.
+     *
+     * @param project
+     * @param variant
+     * @return
+     */
+    static Task getDexTask(Project project, ApplicationVariant variant) {
+        def name = variant.name.capitalize()
+        def rv = project.tasks.findByName("transformClassesWithDexFor${name}")
+        if (rv != null) {
+            return rv
+        }
+        return project.tasks.findByName("dex${name}")
+    }
+
+    /**
+     * Returns the path to the debug meta properties file for the given variant.
+     *
+     * @param project
+     * @param variant
+     * @return
+     */
+    static String getDebugMetaPropPath(Project project, ApplicationVariant variant) {
+        return "${project.rootDir.toPath()}/app/build/intermediates/assets/${variant.dirName}/sentry-debug-meta.properties"
```

Comment:
Did we not find a better way to get this path? I wonder how often people change this stuff...